### ISSUE TITLE
Intellisense not working

### ISSUE THREAD

**Issue by selrond** [silvenon/vscode-mdx#8]
Would it be possible to have intellisense for javascript and markdown inside `.mdx` files? It would require custom rules / config probably. I'm not aware how it works though...

**Comment by silvenon** [comment]
Hopefully. I'm very new to VS Code hacking, but I'll give it a shot some time.

**Comment by chrispaynter** [comment]
Yeah this would be a great addition to the plugin.

I appreciate the effort it would require to make it possible. 

I wish I had free time to contribute to this, as I think intellisense support will become crucial as MDX uptake increases with Storybook. 

I'm already heavily missing it, but the plugin as is has helped my workflow a lot already, so thank you.

**Comment by silvenon** [comment]
Thank you for the kind words, and I totally agree, MDX is going places.

Btw, I just want everyone to know that the inactivity in this repo is only temporary, I'm still heavily using it and I plan to make it much better.

**Comment by piotrpalek** [comment]
Does "intellisense" also include stuff like "import autocomplete"? Currently it's hard to import things into `.mdx` files, because no popup shows up to assist in importing.

**Comment by riceboyler** [comment]
For me, I think v1 of intellisense would be just pulling in things like props, etc. For example, if I do `<Grid justifyContent="" />` using MUI, it'd be nice if it gave me available options for justifyContent.

**Comment by silvenon** [comment]
Yeah, that sucks. If I knew how to add intellisense I would do it right now, but I need to allocate time to level-up in VS Code. The syntax highlighting part was simple, but hooking into these features might require deeper knowledge and possibly some refactoring.

**Comment by remorses** [comment]
The html implementation for script tags could be a starting point
https://github.com/microsoft/vscode/blob/8f08c2ffa75e9ca53ac8ef67ff023feeb94fe5fc/extensions/html-language-features/server/src/modes/javascriptMode.ts


The idea is to convert the current MDX file to a js file, converting everything that is not js to an empty line (this way maintaining the typescript rages correct), then send it to the typescript language server host to get the completion results

**Comment by silvenon** [comment]
Thanks for the tip!

**Comment by karlhorky** [comment]
I guess @remorses tip also mirrors the comment from @mjbvz here: https://github.com/microsoft/vscode/issues/92805#issuecomment-599852139

Another thing mentioned over there is that [`vuejs/vetur`](https://github.com/vuejs/vetur) is also a really good reference for doing this.

And also, that "this won't be easy" :)

**Comment by jasongerbes** [comment]
Hey, any updates on this feature request?

**Comment by adueck** [comment]
Friendly ping to @silvenon 🛎 ... sorry to bug you but wondering if this is still in the works. This would be a huuuge help if it were available. I have a project where I'm writing MDX content with react components that take complicated props. It would be fantastic to get autocomplete / type-checking on the props.

```
This is a blog post with some text... now I'll show you my **interactive component**

<InteractiveThing aPropWhatGoesHereIForget={"help!"} />

```

**Comment by karlhorky** [comment]
Seems like this is also an issue over at `mdx-js/vscode-mdx`, which seems like it could be possibly more regularly maintained:

https://github.com/mdx-js/vscode-mdx/issues/165
